packages/strapi-plugin-users-permissions/controllers/Auth.js

```diff
@@ -251,20 +249,17 @@ module.exports = {
       })
       .get();
 
-    const [protocol, host] = strapi.config.url.split('://');
-    _.defaultsDeep(grantConfig, { server: { protocol, host } });
-
     const [requestPath] = ctx.request.url.split('?');
     const provider =
       process.platform === 'win32' ? requestPath.split('\\')[2] : requestPath.split('/')[2];
-    const config = grantConfig[provider];
 
-    if (!_.get(config, 'enabled')) {
+    if (!_.get(grantConfig[provider], 'enabled')) {
       return ctx.badRequest(null, 'This provider is disabled.');
     }
     // Ability to pass OAuth callback dynamically
-    grantConfig[provider].callback =
-      ctx.query && ctx.query.callback ? ctx.query.callback : grantConfig[provider].callback;
+    grantConfig[provider].callback = _.get(ctx, 'query.callback') || grantConfig[provider].callback;
+    grantConfig[provider].redirect_uri = `${strapi.config.server.url}/connect/${provider}/callback`;
```

Comment:
this should be /auth/${provider}/callback

Reply:
I rewrited what was before. Before, this variable was auto-generated with this value : ![Capture d’écran 2020-05-05 à 15 54 10](https://user-images.githubusercontent.com/11708220/81074204-f445e680-8ee8-11ea-83ce-a5ef9ed6cc03.png)
https://github.com/simov/grant

I haven't digged very much in this topic so maybe it was wrong before and we should change it.
Why do you say it should be `/auth/${provider}/callback` ?